src/WebJobs.Extensions.DurableTask/Options/StorageProviderOptions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Options
+{
+    public class StorageProviderOptions
+    {
+        private CommonStorageProviderOptions configuredProvider;
+
+        public AzureStorageOptions AzureStorage { get; set; }
+
+        public EmulatorStorageOptions Emulator { get; set; }
+
+        internal CommonStorageProviderOptions GetConfiguredProvider()
+        {
+            if (this.configuredProvider == null)
+            {
+                var storageProviderOptions = new CommonStorageProviderOptions[] { this.AzureStorage, this.Emulator };
+                var activeProviders = storageProviderOptions.Where(provider => provider != null);
+                if (activeProviders.Count() != 1)
+                {
+                    throw new InvalidOperationException("Must have one and only one storage provider configured.");
```

Comment:
I would rephrase this slightly: `There must be exactly one storage provider configured.`